department-of-veterans-affairs/caseflow
# HunJerBAH/appeals 4290

Resolves [APPEAL-4290](https://vajira.max.gov/browse/APPEALS-4290)

### Description
Created a Split Appeal button on that renders on the Edit Intake page if the user is a member of the Supervisory Senior Council or Clerk of the Boards, if the appeal has two or more issues, and if the appeal has no pending saves. The Split Appeal button will take the user to the Create Split Appeal page. 

### Acceptance Criteria
A "Split appeal" option is visible on the Edit Issues page when all of the following are true
- [ ] An appeal has more than one issue, AND
- [ ] The BVADispatch Task for the appeal has not been completed, AND
- [ ] User is assigned to one of the following organizations
1. Clerk of the Board
2. Supervisory Senior Council
### The feature toggle for the epic is enabled
- [ ] If one of the conditions defined in A/C 3 are not met, then the Split appeal button will not be visible to the user
- [ ] If the user has added or removed an issue and still needs to save the Split Appeal button is disabled
- [ ] SSC users can always access the correct issues link on the case detail screen

### Testing Plan
1. Load up Caseflow on your Local machine. 
2. Sign in as a SSC user **SPLTAPPLSNOW** by searching Supervisory Senior Council. 
![image](https://user-images.githubusercontent.com/99915461/184450814-b1478b6d-7620-4de8-9da6-7e8f0441c25d.png)

3. Grab a user UUID from the database and navigate to `localhost:3000/appeals/uuid_here/edit`.
4. The split appeal button should **not** originally appear on the page. Add 2 tasks to the appeal. The split appeal button should now show, but be disabled. Click Submit to save the appeal.
![image](https://user-images.githubusercontent.com/99915461/184450400-2ad04bc2-49a1-461a-8e99-70de68bf9794.png)

5.  You should now be on the Queue issues page. Make sure the page contains the **correct issue**s link. Click the **Correct Issues** link to return to the page. 
![image](https://user-images.githubusercontent.com/99915461/184448910-30769efd-97dc-4f18-8c0a-840994ca8400.png)

7. Click the **split appeal** button should now be clickable! 
![image](https://user-images.githubusercontent.com/99915461/184450553-8f0eca62-f677-43e6-9020-2ddbdb1e59b8.png)

8. Click **split appeal**. You should arrive on the create new appeal stream page.
![image](https://user-images.githubusercontent.com/99915461/184450613-3bcf186f-4959-4b3f-b109-2943fdf078d3.png)

Repeat steps 1-8 with a Clerk of the Board user **DOANBVAN**.

9. Make sure the **Split appeal** button does not populate on the **intake/add_issues** page. To do this, first log in as a mail intake admin. 
![image](https://user-images.githubusercontent.com/99915461/184623323-35322ae0-962b-4046-a626-774daee78578.png)

10. navigate to **intake/** on your local machine. Input data in the data fields until you get to the **Add/Remove Issues** page. Make sure the only button on **Add/Remove Issues** is the **Add issue**.
![image](https://user-images.githubusercontent.com/99915461/184623467-14356590-2259-4104-8bbf-d0280bf86b88.png)
![image](https://user-images.githubusercontent.com/99915461/184623494-bbefbd71-deb7-466b-b4dd-ebc40f2e9277.png)
![image](https://user-images.githubusercontent.com/99915461/184623875-c38869dd-5b8b-49a4-94df-794f432b271a.png)
